docs/AdvancedConfiguration.md

```diff
@@ -115,6 +115,57 @@ val columnStats =
      |"date_max":"${formatter.format(maxTimestamp)}" }""".stripMargin
 ```
 
+## String indexing via Histograms
+
+The default **String** column transformation (`HashTransformation`) has limited range query supports since the lexicographic ordering of the String values are not preserved.
+
+This can be addressed by introducing a custom **String** histogram in the form of sorted `Seq[String]`, and can lead to several improvements including:
+1. more efficient file-pruning because of its reduced file-level column min/max
+2. support for range queries on String columns
+3. improved overall query speed
+
+The following code snippet demonstrates the extraction of a **String** histogram from the source data:
+```scala
+ import org.apache.spark.sql.delta.skipping.MultiDimClusteringFunctions
+ import org.apache.spark.sql.DataFrame
+ import org.apache.spark.sql.functions.{col, min}
+
+ def getStringHistogramStr(columnName: String, numBins: Int, df: DataFrame): String = {
```

Comment:
Maybe not for this PR, but would be nice if the user could call this code or that could activate an option to let qbeast-spark build it internally (for example: spark.qbeast.analyze.histogram.enabled)

what do you think?

Reply:
For next steps.